index.js

```diff
@@ -53,7 +54,12 @@ function NYC (opts) {
   // require extensions can be provided as config in package.json.
   this.require = arrify(config.require || opts.require)
 
-  this.transform = this._createTransform()
+  this.extensions = arrify(config.extensions || opts.extension)
```

Comment:
This should access `config.extension` I think. Like `--require` results in `opts.require` and `config.require`, so should `--extension`.